File: StudyHiveAPI/Models/Rating.cs
namespace StudyHiveAPI.Models
{
    public class Rating
    {
        public int Id { get; set; }
        public int Stars { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int MaterialId { get; set; }
        public Material Material { get; set; }
    }
}

File: frontend/lib/models/rating.dart
class Rating {
  final int? id;
  final int materialId;
  final String? userId;
  final int stars;
  final String? userName;

  Rating({
    this.id,
    required this.materialId,
    this.userId,
    required this.stars,
    this.userName,
  });

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      id: json['id'],
      materialId: json['materialId'] ?? 0,
      userId: json['userId'],
      stars: json['stars'] ?? 0,
      userName: json['userName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'materialId': materialId,
      'stars': stars,
    };
  }
}
